### PR TITLE
docs: pwa

### PR DESCRIPTION
**Pull request by mGGk-fr**
# What

This PR adds documentation for PWA setup

# Demo

https://deploy-preview-788--heuristic-almeida-1a1f35.netlify.app/docs/remixed/guides/pwa-setup

### CODE REVIEW

**Review by real34 — APPROVED** [review]
Thanks for this documentation. It looks good :+1:

File: docs/02-guides/pwa-setup.mdx [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+<p>{frontMatter.description}</p>
+
+## What is a PWA
+
+PWA stands for Progressive Web App, it will allow your shop to be installed as
+an application on end user devices (mobile, desktop, tablets).
+
+It allow advanced caching on device, fastest navigation, offline support and
+much more.
```
Comment (dpobel) [inline docs/02-guides/pwa-setup.mdx:17]:
```suggestion
[PWA stands for Progressive Web App](https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps). This technology allows the user to install your store as a native application on their device. It also brings advanced caching capabilities, offline support and much more.
```

File: docs/02-guides/pwa-setup.mdx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+Front-Commerce integrates PWA capabilities and is available with a few
+configuration.
+
+## Setup PWA
+
+### `webmanifest` configuration
+
+The [`webmanifest`](https://developer.mozilla.org/en-US/docs/Web/Manifest) will
+define meta informations about your store. To setup this informations, you needs
+to add the following configuration in your `front-commerce.config.ts` file:
```
Comment (dpobel) [inline docs/02-guides/pwa-setup.mdx:28]:
```suggestion
The [`webmanifest`](https://developer.mozilla.org/en-US/docs/Web/Manifest)
defines meta information about your store. To setup this webmanifest, you need
to add the following configuration in your `front-commerce.config.ts` file:
```

File: docs/02-guides/pwa-setup.mdx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    themeColor: "#fbb03b",
+    icon: "public/pwa-icon.png",
+    maskableIcon: "public/pwa-masked-icon",
+    ttl: 60,
+  },
+  // highlight-end
+});
+```
+
+Here's a description of options that can be configured:
```
Comment (dpobel) [inline docs/02-guides/pwa-setup.mdx:57]:
```suggestion
Here is a list of the available options:
```

File: docs/02-guides/pwa-setup.mdx [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+| `shortName`    | `string` | The short name of your PWA, it will be displayed on the home screen of your customer                                 |
+| `themeColor`   | `string` | The theme color of your PWA, it will define theme for your App, we recommend you to use your website primary color   |
+| `icon`         | `string` | The icon of your PWA, it will be displayed on the home screen of your end user device, it must be at least 512x512   |
+| `maskableIcon` | `string` | The maskable icon of your PWA, it will be displayed on the home screen of your customer, it must be at lease 512x512 |
+| `ttl`          | `number` | The time to live of your PWA Cache in milliseconds, the default value is of 7 days                                   |
+
+### Icons setup
+
+PWA require multiple icons to be defined in your project that will be used by
+device to identify your PWA on end-user device screen.
```
Comment (dpobel) [inline docs/02-guides/pwa-setup.mdx:71]:
```suggestion
A PWA requires multiple icons to be defined in your project that will be used by
the device to identify the application on the end-user device screen.
```

File: docs/02-guides/pwa-setup.mdx [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+## Install PWA
+
+Once everything is setup, you will be able to install your website as an
+application.
+
+:::caution
+
+Except when on localhost, PWA installation is only possible with **HTTPS**. If
+you're using Front-Commerce Cloud, HTTPS is enabled by default. If you use
+you're own server, you need to enable HTTPS.
```
Comment (dpobel) [inline docs/02-guides/pwa-setup.mdx:113]: not sure we need to add that in 2023 :)

File: docs/02-guides/pwa-setup.mdx [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+
+:::
+
+Depending on the browser you're using, the browser may display a prompt to
+invite you to install the application. Otherwise, please check your browser
+documentation on how to install PWA.
```
Comment (dpobel) [inline docs/02-guides/pwa-setup.mdx:119]:
```suggestion
Depending on the browser, a prompt to
invite the customer to install the application is different:
```

File: docs/02-guides/pwa-setup.mdx [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+
+- Chrome: https://support.google.com/chrome/answer/9658361
+- Safari : https://web.dev/learn/pwa/installation/#ios-and-ipados-installation
+- Firefox :
+  https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Guides/Installing
+- Edge :
+  https://learn.microsoft.com/en-us/microsoft-edge/progressive-web-apps-chromium/ux
```
Comment (dpobel) [inline docs/02-guides/pwa-setup.mdx:126]:
```suggestion
- Safari: https://web.dev/learn/pwa/installation/#ios-and-ipados-installation
- Firefox:
  https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Guides/Installing
- Edge:
  https://learn.microsoft.com/en-us/microsoft-edge/progressive-web-apps-chromium/ux
```

**Review by dpobel — APPROVED** [review]

File: docs/02-guides/pwa-setup.mdx (hunk 1 above)
Comment (mGGk-fr) [reply docs/02-guides/pwa-setup.mdx]: fixed with your suggestion

File: docs/02-guides/pwa-setup.mdx (hunk 2 above)
Comment (mGGk-fr) [reply docs/02-guides/pwa-setup.mdx]: fixed with your suggestion

File: docs/02-guides/pwa-setup.mdx (hunk 3 above)
Comment (mGGk-fr) [reply docs/02-guides/pwa-setup.mdx]: fixed with your suggestion

File: docs/02-guides/pwa-setup.mdx (hunk 4 above)
Comment (mGGk-fr) [reply docs/02-guides/pwa-setup.mdx]: fixed with your suggestion

File: docs/02-guides/pwa-setup.mdx (hunk 5 above)
Comment (mGGk-fr) [reply docs/02-guides/pwa-setup.mdx]: removed

File: docs/02-guides/pwa-setup.mdx (hunk 6 above)
Comment (mGGk-fr) [reply docs/02-guides/pwa-setup.mdx]: fixed with your suggestion

File: docs/02-guides/pwa-setup.mdx (hunk 7 above)
Comment (mGGk-fr) [reply docs/02-guides/pwa-setup.mdx]: fixed with your suggestion

**Pull request merged**